### PR TITLE
Upgrade dartdoc to 0.15.0+1

### PR DESCRIPTION
**Pull request by jcollins-g**
Changes:

- 0.15.0+1: https://github.com/dart-lang/dartdoc/releases/tag/v0.15.0%2B1
- 0.15.0: https://github.com/dart-lang/dartdoc/releases/tag/v0.15.0

Sample docs:
- 0.14.1: http://jcollins1.pdx.corp.google.com:8000
- 0.15.0+1: http://jcollins1.pdx.corp.google.com:8001

Highlights:

Reliably hidden private classes in inheritance trees:  http://jcollins1.pdx.corp.google.com:8001/flutter/dart-collection/ListQueue-class.html 
Properly resolved abstract types in methods/properties (see "distinct"):  http://jcollins1.pdx.corp.google.com:8001/flutter/dart-io/Socket-class.html 
Explicitly defined setters no longer leak parameter names into property summaries:  
http://jcollins1.pdx.corp.google.com:8001/flutter/semantics/SemanticsConfiguration-class.html
Included source code pretty-prints correctly again:
http://jcollins1.pdx.corp.google.com:8001/flutter/semantics/SemanticsConfiguration/isButton.html 
All libraries in dependent packages are documented with --auto-include-dependencies, rather than just the libraries used by the top level package:
http://jcollins1.pdx.corp.google.com:8001/index.html 
Anonymous libraries are laid out differently on disk to avoid conflicts:
http://jcollins1.pdx.corp.google.com:8001/flutter/package-file_memory/package-file_memory-library.html

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -109,7 +109,10 @@ dependencies:
     '--header', 'analytics.html',
     '--header', 'survey.html',
     '--footer-text', 'lib/footer.html',
-    '--exclude', 'temp_doc',
+    '--exclude-packages',
+'analyzer,args,barback,cli_util,csslib,front_end,glob,html,http_multi_server,io,isolate,js,kernel,logging,mime,mockito,node_preamble,plugin,shelf,shelf_packages_handler,shelf_static,shelf_web_socket,utf,watcher,yaml',
```
Comment (Hixie) [inline dev/tools/dartdoc.dart]: i'm surprised to see `io` and `plugin` in this list, but I'm not sure exactly which those refer to... is it dart:io and something to do with flutter plugins, or something else?

**Comment by Hixie** [comment]
LGTM

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (jcollins-g) [reply dev/tools/dartdoc.dart]: These refer to: https://pub.dartlang.org/packages/io and https://pub.dartlang.org/packages/plugin, both included in the transitive package dependencies.  

The exclude-packages line was constructed to minimize the change with the new recursive package handling.  We can change it; the current value was constructed mechanically.

**Comment by sethladd** [comment]
Very cool!

**Pull request merged**